File: CS50/pset6/credit/credit.py
from cs50 import get_int

input = get_int("Number: ")
# the original input
orinput = input

length = len(str(input))
# if this number is even add to evenTtl else to oddTtl, it is also used as a number counter
count = 0
evenTtl = 0
oddTtl = 0

valid = False

for i in range(length):

    count += 1
    number = input % 10

    if count % 2 == 0:
        number *= 2

        if number > 9:
            number -= 9

        evenTtl += number

    else:
        oddTtl += number

    input = input // 10

if (evenTtl + oddTtl) % 10 == 0:
    valid = True
else:
    print("INVALID")


# AMERX 15 STRT 34 OR 37
# MC 16 STRT 51, 52, 53, 54, 55
# VZA 13 OR 16 STRT 4

if valid == True:
    if (orinput >= 4 * (10 ** 12) and orinput < 5 * (10 ** 12)) or (orinput >= 4 * (10 ** 15) and orinput < 5 * (10 ** 15)):
        print("VISA")
    elif (orinput >= 34 * (10 ** 13) and orinput < 35 * (10 ** 13)) or (orinput >= 37 * (10 ** 13) and orinput < 38 * (10 ** 13)):
        print("AMEX")
    elif orinput >= 51 * (10 ** 14) and orinput < (56 * (10 ** 14)):
        print("MASTERCARD")
    else:
        print("INVALID")

File: CS50/credit.c
#include <stdio.h>
#include <cs50.h>


long power();


int main(void)
{
    //AMERX 15 STRT 34 OR 37
    //MC 16 STRT 51, 52, 53, 54, 55
    //VZA 13 OR 16 STRT 4

   long input;
   bool isValid = true;
   string type;
   int mod = 10;
   int place = 1;
   int num = 0;
   int totalEven = 0;
   int totalOdd = 0;

   do
   {
      input = get_long("Number: ");
      

   }

   while(input < 0);
   
      
      for(int i = 0; i < 16; i++)
      {
          num = ((input % mod) - (input % (mod /10))) / (mod/10);
          if(place % 2 == 0)
          {
              totalEven = totalEven + num * 2;
          }
          else
          {
             totalOdd = totalOdd + num; 
          }
          
          mod = mod * 10;
          place++;
      }
      
      if((totalEven + totalOdd) % 10 == 0 )
      {
          isValid = true;
      }
      else
      {
          isValid = false;
          
          printf("%i , %i", totalEven, totalOdd);
      }
   if (isValid == true){
   if((input < (38 * power(10, 13)) && input >=(37 * power(10, 13))) || (input < (35 * power(10,13)) && input >= (34 * power(10, 13)))) 
   {
       type = "AMEX\n";
   }
   else if(input >= (51 * power(10, 14)) && input < (56 * power(10, 14)))
   {
       type = "MASTERCARD\n";
   }
   else if((input < (5 * power(10, 12)) && input >= (4 * power(10, 12))) || (input < (5 * power(10, 15)) && input >= (4 * power(10, 15))))
   {
       type = "VISA\n";
   } 
       else{
       type = "error\n";
   }
}
   else
   {
       type = "INVALID\n";
   }
   

    if((totalEven + totalOdd) % 10 == 0 )
    {
      isValid = true;
    }
    else
    {
      isValid = false;
    }
      
    printf("%s", type);

}





long power(int n, int p)
{
    long result = 1;
    for(int i = 0; i<p; i++)
    {
        result = result * n;
    }
    return result;